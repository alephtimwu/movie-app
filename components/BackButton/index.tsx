import React from 'react';
import { Image, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  Back: {
    width: 20,
    height: 11,
    marginLeft: 25
  }
});

const BackButton = () => (
  <Image
    source={require('../../assets/icon-arrow-left.png')}
    style={styles.Back}
  />
);

export default BackButton;
