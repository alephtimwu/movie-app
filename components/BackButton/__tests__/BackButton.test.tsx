import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
// import { CrpAll } from '__mocks__';
import UseAppContext from '../../../store/index';
import BackButton from '../index';

describe('BackButton page =>', () => {
  let tree = null;

  beforeEach(() => {
    tree = render(<BackButton />);
  });

  test('BackButton result page should render correctly', async () => {
    expect(tree).toMatchSnapshot();
  });

  // test('Click on footer back', () => {
  //   const { getAllByText } = tree;
  //   const [back] = getAllByText('Back to Dashboard');
  //   fireEvent.click(back);
  // });
  // test('Click on Have us contact you', () => {
  //   const { getAllByText } = tree;
  //   const [contact] = getAllByText('Have us contact you');
  //   fireEvent.click(contact);
  // });
  // test('Click on retake', () => {
  //   const { getAllByText } = tree;
  //   const [retake] = getAllByText('retake');
  //   fireEvent.click(retake);
  // });
});
