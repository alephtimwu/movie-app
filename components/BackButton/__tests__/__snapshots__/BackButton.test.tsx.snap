// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton page => BackButton should render correctly 1`] = `
<Image
  source={1}
  style={
    Object {
      "height": 11,
      "marginLeft": 25,
      "width": 20,
    }
  }
/>
`;
